app/controllers/advertisements_controller.rb

```diff
@@ -48,13 +50,24 @@ def is_admin_or_owner
 
     if  @advertisement.nil? or @advertisement.user_id.nil?
       render status: :not_found
-    elsif is_admin? || @advertisement.user_id == current_user.id
+    elsif is_admin? #|| @advertisement.user_id == current_user.id
       return
     else
       render json: { message: "Access forbidden." }, status: :forbidden
     end
   end
 
+  def is_owner
```

Comment:
normally methods should be as small as possible, and they have to do only one thing (see [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle))

Reply:
Updated _is_owner_ method and removed is_admin. Replaced is_admin? method into user model. Updated methods for update and destroy in comments ans adverts controllers.